Tween/SequencedAnimation.cs

```diff
@@ -25,7 +25,7 @@ protected override void PlayQueuedAnimations()
 				{
 					nextAnimation.Play(HandleCurrentAnimationCompleted, () =>
 					{
-						if (!nextAnimation.IsValid)
+						if (IsPlaying && !nextAnimation.IsValid)
```

Comment:
I can see the argument for wanting a `ParalleledAnimation` to continue if one part is invalid, but I don't think a `SequencedAnimation` ever should, especially not silently. I'd prefer if both collections Abort if one part fails, I can't think of any situations that would be good design for it to happen intentionally. At the very least if they do continue, I'd like to know about it with a `Debug.LogWarning`.

Reply:
They should continue -- or the whole validation check system will no longer make sense. I added the `IsValid` check for reasons. It gives fewer crashes and in general, improves the stability of the program. For a released production, the very last thing you wanted to see is crashes.

I had the compiler flag idea for a couple of years. Which you define a flag for your build i.e. `#PRODUCTION`, then the validation will do every silently and not to crashing your game. But without the flag, it will crash and help you debug your game. This idea is very hard to apply since DUCK contains so many gadgets and I simply can't add a random flag for an open source library.

For now, I'll add a warning message there.

Reply:
Before I added the validation system -- which is the lazy system that helps the developers maintain their invalid animations. The `Fancy™Tween` animation system has been complained by people quite often that it crashes a lot.

Then I decided, instead of teaching people how to use this system correctly and how to write complicated code to maintain your animations, I made this validation system.

It's like the Garbage Collection system. You can ignore it, for a while, until you hit the limit. Then you need to code properly and always keep this thing in your mind. Same for this `Fancy™Tween`. You can use it in a very rough way and it still works -- but it doesn't mean you have done it in its very best form. In your case, you integrated the animation system deeply into your game and you starting notice issues. I believe by now we are kinda dug out every potential of the `Fancy™Tween` system. And it's time to think about some hardcore usages and meanwhile, not losing its simplicity for entry-level usages. This is very hard to do so.

Reply:
I see where you're coming from and I agree when it's a simple self contained part you're making safer, but some of the uses we're covering are really hard to debug if they're covered up. E.g. animation collections of various classes extending different types of animation, looping options, things hanging depending on callbacks, and the fact that they're all just added to the update function of one update list make it really hard to debug and actually find the source of a problem.

Also a lot of the time with the collections I don't think we'll actually prevent an exception, just delay it or even cause more in this case. Quite often someone will use an animation collection and in the `onComplete` call a method on one of the targets, set the gameobject inactive etc. If the collection silently skips an invalid part in the middle but still calls the `onComplete`, this will now throw whereas if we aborted the whole animation collection it wouldn't. Unless we are now expected to null check every object we use in the callback, but that adds a lot more code and defeats part of the purpose of the `IsValid` stuff anyway.

Reply:
For me it boils down to this question "how many errors do we want to swallow?"

Chuan quite rightly pointed out that you don't want your production build to crash, but also you definitely do want your dev builds to error. I don't want to mess about with compile flags either so the question we now ask is, Would you rather your production build fails silently, or your dev build is very explicit. I think I prefer explicity here.

Dan mentioned, that even if you attempt to continue after a failure, you are likely just prolonging the failure, since your behaviour is incorrect. You can leave an object in the incorrect state. There is a tendency to think "oh it's just an animation, it's not that important". But the way we use animations at current is very coupled to the logic and flow of the system. The incorrect behaviour can actually cause bugs like not fading out a loading screen, leaving something blocking something else.

None of us want our production games to crash, but the chances are if you are very explicit with the developer during the course of development there are less likely to be hidden bugs in the release. If we swallow errors, even providing warnings it's very easy to ignore them, and then you forget until it causes a hidden bug IN production that is impossible to debug.

I see both sides. but I lean towards throwing, As for auto reference, I think if an animation becomes invalid (due to destroying an object). Just throw and tell them to fix it. You will be surprised how quickly people will learn what is correct and incorrect usage of the system.